src/main/java/org/apache/maven/archiver/MavenArchiver.java

```diff
@@ -800,6 +800,21 @@ public static Optional<Instant> parseBuildOutputTimestamp(String outputTimestamp
         }
     }
 
+    private static boolean isNumeric(String str) {
+
+        if (str.isEmpty()) {
+            return false;
+        }
+
+        for (char c : str.toCharArray()) {
+            if (!Character.isDigit(c)) {
```

Comment:
This is likely not what you want. It includes lots of non-ASCII digits like Devanagari digits. Long.parseLong won't work on these.

There might also be an issue with negative numbers.

Reply:
It is simple copy method from plexu-utils ...
https://github.com/codehaus-plexus/plexus-utils/blob/master/src/main/java/org/codehaus/plexus/util/StringUtils.java#L1749

I think it is enough for our case - if user provide a special chars Long.parseLong simply throw exception if needed.

Reply:
The same implementation is used in common-lang and commons-lang3

Reply:
I guess this is within spec for parseBuildOutputTimestamp, as long as it's private and not used anywhere else.